### ISSUE TITLE
Slackbot - call for discussion, then development

### ISSUE THREAD

**Issue by knewter** [dailydrip/firestorm#24]
We need to build a slackbot once the core forum is chugging along nicely, because we hit that stretch goal! If you have thoughts about the overall design, go ahead and spam them in this issue. We'll collect them up and have a design discussion around them once 1.0 launches, before beginning development (which will also be done on this issue).

**Comment by SoreGums** [comment]
What's the point of the bot? Or is that what this issue is for ?

I'd like there to be a way to share roles. If there is a category in firestorm and a group has access to it and there is a channel in discord with a specific role. Then a user who has connected their discord account to their firestorm account should get access to the firestorm categories that match their discord roles and vice versa.

The next thing is you're able to assign flair or something to people on discord so having that automatically sync and be visible in firestorm would be ideal too.

Once firestorm is done I'm hoping I'll be able to use it as the forum part of an information system I have planned. Our group uses another forum software, however I don't like it as it takes forever to load (1-2secs lol) after each click and the screen always shuffles itself around while leading.

So this sync between discord and firestorm user and groups/roles would be great + the flair/attribute. If that is not what is intended, then no biggie, I'm sure it won't be difficult to do it on my own.

**Comment by Trevoke** [comment]
What are the things a bot can do, as intermediary between Slack and a forum?

It could notify:
- when a new post appears in a category
- when a reply is made to a particular post 
- when a reply is made to all posts in a category
- of changes to a poll (assuming we have things like polls)
- of the status, basic metrics, whatever (100 emails sent per hour, 4 new posts per minute, etc)

It could be used...
- to start a new post, maybe from a Slack message
- to convert a Slack message into a reply to a thread
- to vote on a poll
- to ask for personal stats
- to send someone a DM on the forum
- to retrieve a particular topic that is relevant to the conversation
- to search for topics

... What else?

**Comment by alanwilhelm** [comment]
I do something like this in another app. https://github.com/BlakeWilliams/Elixir-Slack is working well. I use it to notify via slack for a large number of system events and also to perform system actions from slack. Is that library an option?

**Comment by knewter** [comment]
@alanwilhelm I like the idea of using https://github.com/hedwig-im/hedwig. I'm super familiar with it and it has the benefit that a slackbot becomes an anything-else bot trivially just by adding adapters

**Comment by alanwilhelm** [comment]
@knewter that makes sense, way more flexible!
